src/main/java/ca/team2706/frc/robot/subsystems/DriveBase.java

```diff
@@ -631,6 +633,14 @@ public void resetEncoders() {
         rightFrontMotor.getSensorCollection().setQuadraturePosition(0, Config.CAN_SHORT);
     }
 
+/**
+     * Resets the gyro to 0 degrees
+     */
+    public void initGyro() {
+        savedAngle = Config.getROBOT_START_ANGLE();
+        gyro.setYaw(0, Config.CAN_SHORT);
```

Comment:
Is there a reason that you don't just initialize `savedAngle` to `Config.ROBOT_START_ANGLE.get()` after calling reset in the constructor? If you want to be able to reset the absolute heading at a later point, you could also add a `resetAbsoluteHeading()` method.

Reply:
By the way, I want to mention that the javadoc for the initGyro() method was not correct and has been corrected.

First, I want to make sure that my understanding of the gyro and associated code is correct so I'll explain what I think is going on so you can correct me if I'm wrong. 

Ultimately, we want to be able to get the absolute heading of the robot where the heading of the robot is measured relative to a horizontal line across the field where the robot pointing horizontally to the right at zero degrees and the heading increases positively in the CCW direction. 


getAbsoluteHeading() is given by

```
public double getAbsoluteHeading() {
        return savedAngle + getHeading();
}
```
and returns savedAngle + getHeading(), where getHeading() returns a relative angular measurement from the gyro. This relative angular measurement is likely zero when the robot is powered on but just to be safe we probably want to reset it to zero at startup before the robot has moved just so we know were starting from a known position.

In order for the robot to provide an absolute heading as defined above, it must be "told" what its current heading is when it is powered on. The is done through Config.ROBOT_START_ANGLE angle the robot must be physically placed at that heading when it is powered on.

The initGyro() method is called from the constructor only at startup when the robot is presumably at the heading Config.ROBOT_START_ANGLE. The initGyro() method is given by

```
public void initGyro() {
        savedAngle = Config.getROBOT_START_ANGLE();
        gyro.setYaw(0, Config.CAN_SHORT);
}
```
so that savedAngle will be set to Config.ROBOT_START_ANGLE, the relative gyro will be set to zero, and the value returned by getAbsoluteHeading() will be Config.ROBOT_START_ANGLE and thus as expected. As the robot moves, getAbsoluteHeading() will continue to provide the correct absolute heading.

At various times, there is a need to reset the relative gyro to zero degrees. (I actually don't know why this is necessary, but I'll accept it as a fact of life.) When this happens, we don't want to lose the ability to provide the absolute heading. Therefore, to reset the relative gyro, we call resetGyro() which is

```
public void resetGyro() {
    savedAngle = getAbsoluteHeading();
    gyro.setYaw(0, Config.CAN_SHORT);
}
```
which will store the current absolute heading in savedAngle and then reset the relative gyro to zero so that calling getAbsoluteGyro() after that will continue to provide the correct absolute heading.

When we tested the above approach on Plyboy, the measured results were as expected.

Is this the way it is supposed to work? Let me know if you would like further changes.

Reply:
That makes more sense. Maybe `resetAbsoluteGyro()` would be a better name for it, so it matches the name of the other method.